HLint.hs

```diff
@@ -3,6 +3,8 @@ import "hlint" HLint.Dollar
 import "hlint" HLint.Generalise
 import "hlint" HLint.HLint
 
+ignore "Avoid lambda" = Action.mkRunAction
```

Comment:
> > +ignore "Avoid lambda" = Action.mkRunAction
>
> If we use `(Nat . run)` instead of `\ s -> Nat (run s)` the type checker misses the `forall a` quantification, which is needed for the Natural transformation.

I'm ok with this, but I'm wonderung: could we leave "Avoid lambda" enabled and add an explicit type signature there?